cryptoprotocol/src/main/java/org/cryptimeleon/incentivesystem/cryptoprotocol/IncentiveSystem.java

```diff
@@ -241,19 +242,24 @@ public EarnRequest generateEarnRequest(Token token, ProviderPublicKey providerPu
     public SPSEQSignature generateEarnRequestResponse(EarnRequest earnRequest, BigInteger k, ProviderKeyPair providerKeyPair) {
 
         // Verify the blinded signature for the blinded commitment is valid
-        var isSignatureValid = pp.getSpsEq().verify(providerKeyPair.getPk().getPkSpsEq(), earnRequest.getBlindedSignature(), earnRequest.getC0(), earnRequest.getC1());
+        var isSignatureValid = pp.getSpsEq().verify(
+                providerKeyPair.getPk().getPkSpsEq(),
+                earnRequest.getBlindedSignature(),
+                earnRequest.getC0(),
+                earnRequest.getC1()
+        );
         if (!isSignatureValid) {
             throw new IllegalArgumentException("Signature is not valid");
         }
 
         // Sign a blinded commitment with k more points
         var C0 = earnRequest.getC0();
         var C1 = earnRequest.getC1();
-        var q4 = providerKeyPair.getSk().getQ().get(4);
+        var q5 = providerKeyPair.getSk().getQ().get(4); // DLOGs have 1-based indexing in paper, 0-based indexing in code
```

Comment:
I would add something like
... therefore ```.get(4)``` is used for ```q5```

Reply:
Done.